smithery-mcp-server/src/services/document-ingestion.service.ts

```
@@ -0,0 +1,163 @@
+    try {
+        console.log(`[DocIngestion] Attempt 1: Scraping with Gemini URL grounding for ${url}`);
+        const model = genAI.getGenerativeModel({ model: 'gemini-1.5-flash' });
+
+        const prompt = `Please act as a web scraper. Fetch the content from the provided URL and return the full, clean Markdown representation of the main content. Also, provide the document's primary title.\n\nURL: ${url}\n\nRespond in a JSON format with two keys: "title" and "markdownContent".`;
+
+        const result = await model.generateContent(prompt);
+        const responseText = result.response.text();
+
+        // Attempt to parse the JSON from the response, which might be in a markdown block
+        const jsonMatch = responseText.match(/```json\n([\s\S]*?)\n```/);
+        if (jsonMatch && jsonMatch[1]) {
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)